draft-rescorla-tls-ctls.md

```diff
@@ -121,38 +121,62 @@ with a varint.
 
 ## Record Layer
 
-The cTLS Record Layer assumes that records are externally framed
-(i.e., that the length is already known because it is carried in a UDP
-datagram or the like). Depending on how this was carried, you might
-need another byte or two for that framing. Thus, only the type byte
-need be carried and TLSPlaintext becomes:
+The only cTLS records that are sent in plaintext are handshake records
+(ClientHello, and ServerHello/HRR). The content type is therefore constant (it
+is always `handshake`) and does not need to be explicitly conveyed in the record
+layer. Length information is signaled for unencrypted records using the varint
+encoding. (In principle, even this length could be omitted, since handshake
+messages are self-describing. Having this length field allows a clean separation
+between the record layer and the handshake layer.)
 
 ~~~~
       struct {
-          ContentType type;
-          opaque fragment[TLSPlaintext.length];
+          opaque fragment<0..V>;
       } TLSPlaintext;
 ~~~~
 
-In addition, because the epoch is known in advance, the
-dummy content type is not needed for the ciphertext, so
-TLSCiphertext becomes:
+Encrypted records use DTLS 1.3 record framing, comprising a configuration octet
```

Comment:
How are you expecting to distinguish encrypted and unencrypted records? Because the first byte does not have 0x20 set?

Reply:
The only unencrypted records are ClientHello and ServerHello, which are the first records in each direction.  Once you consume those, you switch over to this framing.

Reply:
That doesn't work for DTLS because you might get a retransmission.